### PR TITLE
DeveloperGuide: update on documentation

### PR DESCRIPTION
**Pull request by chyeo**
The current DeveloperGuide has very little content regarding the various design considerations that was made while developing this application 

Let's add in the design considerations for the 'requirement_add' command into the DeveloperGuide

### CODE REVIEW

**Review by Letm3through — APPROVED** [review]
Thank you for the update! Looking forwards to further refinements and adding of sequence diagram in future.

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -347,6 +347,42 @@ corequisites. The user will have to delete the invalid co-requisite manually aft
 refreshed to display the full listing just to be able to iterate and delete modules co-requisites accordingly.
 // end::corequisites[]
 
+// tag::requirement-add-command[]
```
Comment (Letm3through) [inline docs/DeveloperGuide.adoc]: I think can remove command and just put as `requirement-add`

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -347,6 +347,42 @@ corequisites. The user will have to delete the invalid co-requisite manually aft
...
+===== Aspect: Choice of what is added to the requirement category
+
+- **Alternative 1 (current choice):** Storing module codes only
+  ** Pros: Requires less storage, easy to maintain
+  ** Cons: Requires extra overhead to obtain additional information related to the module when needed
+- **Alternative 2:** Storing all information related to the module
+  ** Pros: All information related to the module are easily obtainable
+  ** Cons: Information stored is duplicated, additional storage space is needed
+  ** Cons: Hard to maintain information, if a module information is updated have to update data in the requirement category
+//end::requirement-add-command[]
```
Comment (Letm3through) [inline docs/DeveloperGuide.adoc]: Could you change this accordingly to the above?

**Review by Letm3through — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (chyeo) [reply docs/DeveloperGuide.adoc]: Roger, thanks for the inputs.

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (chyeo) [reply docs/DeveloperGuide.adoc]: Roger, will change accordingly.

**Review by Letm3through — APPROVED** [review]
LGTM. Approving and merging

**Pull request merged**